### PR TITLE
gstreamer: Prefix-match any generic supported codec against the requested codec.

### LINKED ISSUE #46259: canPlayType() rejects valid codec strings: GStreamer registry scanner compares wildcard entries exactly
### Description

`HTMLMediaElement.canPlayType()` returns `""` for valid, playable MIME types that include a `codecs` parameter, e.g.:

```js
video.canPlayType('video/mp4')                                   // "maybe"  ✓
video.canPlayType('video/mp4; codecs="avc1.42001E, mp4a.40.2"')  // ""       ✗ (expected "probably")
```

This makes video.js-based players (Invidious, many news sites) fail with **"No compatible source was found for this media"** before ever attempting playback, even though the GStreamer pipeline plays the same source without issue when `src` is set directly.

### Root cause

In `components/media/backends/gstreamer/registry_scanner.rs`, the scanner inserts *wildcard* codec entries:

```rust
self.supported_codecs.insert("mp4a*");
...
self.supported_codecs.insert("avc*");
self.supported_codecs.insert("mp4v*");
```

but membership is tested with exact `HashSet::contains`:

```rust
fn is_codec_supported(&self, codec: &str) -> bool {
    self.supported_codecs.contains(codec)
}
```

A real codec string like `avc1.42001E` never equals `avc*`, so `are_all_codecs_supported` returns `false` and `can_play_type` (in `backends/gstreamer/lib.rs`) returns `SupportsMediaType::No` for any MIME type carrying codec parameters.

WebKit's `GStreamerRegistryScanner`, which this design follows, treats a trailing `*` as a prefix wildcard when matching codecs.

### Suggested fix

```rust
fn is_codec_supported(&self, codec: &str) -> bool {
    self.supported_codecs.iter().any(|supported| {
        match supported.strip_suffix('*') {
            Some(prefix) => codec.starts_with(prefix),
            None => *supported == codec,
        }
    })
}
```

(Case-insensitivity might also be worth considering; codec strings are commonly lowercase but not guaranteed.)

### Steps to reproduce

1. Build an embedder (or servoshell) with the `media-gstreamer` feature.
2. Load any Invidious watch page, or evaluate the two `canPlayType` calls above on a page with a `<video>` element.
3. Observe `""` for the codec-qualified type; video.js consequently rejects all `<source>` elements.

Tested at rev fe17152d680873e1fa1e11004261365e04765950 on Windows 11 (GStreamer 1.26.11 MSVC). Working around it in my embedder by shimming `canPlayType` via a user script confirms playback works once source selection succeeds.

### PR DESCRIPTION
**Pull request by jdm**
We store values like "mp4*" and perform an exact match against it, but the actual codec values are different (`mp4v.20.8`). This change matches the behaviour of WebKit's codec lookup as well: https://searchfox.org/wubkat/rev/4bf9d2cf94e5a07bdcb977203ddc27f3752ecaf7/Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp#844

Testing: New passing tests.
Fixes: #46259

### CODE REVIEW

File: tests/wpt/webgl/meta/conformance/textures/misc/texture-corner-case-videos.html.ini [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[texture-corner-case-videos.html]
+  [WebGL test #6]
```
Comment (jdm) [inline tests/wpt/webgl/meta/conformance/textures/misc/texture-corner-case-videos.html.ini:2]: These webgl tests call canPlayType and skip the rest of the test if it reported the mp4 video was not playable, so these failures represent progress.

**Review by jschwe — APPROVED** [review]
